src/NuGet.Clients/NuGet.CommandLine/Commands/ClearCacheCommand.cs

```diff
@@ -0,0 +1,140 @@
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+using NuGet.Configuration;
+
+namespace NuGet.CommandLine.Commands
+{
+    [Command(typeof(NuGetCommand), "clearcache", "ClearCacheCommandDescription", MaxArgs = 0,
+        UsageSummaryResourceName = "ClearCacheCommandSummary",
+        UsageExampleResourceName = "ClearCacheCommandExamples")]
+    public class ClearCacheCommand
+        : Command
+    {
+        [Option(typeof(NuGetCommand), "ClearCacheCommandClearGlobalPackagesDescription")]
+        public bool ClearGlobalPackages { get; set; }
+
+        public override Task ExecuteCommandAsync()
+        {
+            int commandResult = 0;
+
+            // Clear the NuGet machine cache
+            if (!string.IsNullOrEmpty(MachineCache.Default?.Source))
+            {
+                Console.WriteLine(
+                    LocalizedResourceManager.GetString(nameof(NuGetResources.ClearCacheCommand_ClearingNuGetCache)),
+                    MachineCache.Default.Source);
+
+                commandResult = ClearCacheDirectory(MachineCache.Default.Source);
+            }
+
+            // Clear NuGet v3 HTTP cache
+            var localAppDataFolderPath = Environment.GetFolderPath(Environment.SpecialFolder.LocalApplicationData);
+            var httpCacheFolderPath = Path.Combine(localAppDataFolderPath, "NuGet", "v3-cache");
+            if (!string.IsNullOrEmpty(httpCacheFolderPath))
+            {
+                Console.WriteLine(
+                    LocalizedResourceManager.GetString(nameof(NuGetResources.ClearCacheCommand_ClearingNuGetHttpCache)),
+                    httpCacheFolderPath);
+
+                var result = ClearCacheDirectory(httpCacheFolderPath);
+                if (commandResult == 0)
+                {
+                    commandResult = result;
+                }
+            }
+
+            if (ClearGlobalPackages)
+            {
+                // also clear the global NuGet packages cache
+                var globalPackagesFolderPath = SettingsUtility.GetGlobalPackagesFolder(Settings);
+
+                Console.WriteLine(
+                    LocalizedResourceManager.GetString(nameof(NuGetResources.ClearCacheCommand_ClearingNuGetGlobalPackagesCache)),
+                    globalPackagesFolderPath);
+
+                var result = ClearCacheDirectory(globalPackagesFolderPath);
+                if (commandResult == 0)
+                {
+                    commandResult = result;
+                }
+            }
+
+            if (commandResult != 0)
+            {
+                throw new CommandLineException(LocalizedResourceManager.GetString(nameof(NuGetResources.ClearCacheCommand_CacheClearFailed)));
+            }
+
+            return Task.FromResult(commandResult);
+        }
+
+        private int ClearCacheDirectory(string folderPath)
```

Comment:
Addressed in https://github.com/NuGet/NuGet.Client/commit/2393072e082b4712cdef3904f52be24da66b60ba